modMcf/src/org/aion/mcf/trie/JournalPruneDataSource.java

```diff
@@ -85,58 +90,109 @@ public JournalPruneDataSource(IByteArrayKeyValueDatabase src) {
     }
 
     public void setPruneEnabled(boolean e) {
+        lock.writeLock().lock();
+
         enabled = e;
+
```

Comment:
use atomicboolean for the variable - enabled, then don't need lock in this method also check enabled before the other methods before writelock.